packages/components/src/internal/components/editable/actions.ts

```diff
@@ -116,75 +94,107 @@ export const loadEditorModelData = async (
 
     return {
         cellValues,
-        columns: List(columns.map(col => col.fieldKey)),
+        orderedColumns: List(columns.map(col => col.fieldKey)),
         deletedIds: ImmutableSet<any>(),
         rowCount: orderedRows.length,
     };
 };
 
-export const initEditableGridModel = async (
-    dataModel: QueryModel,
-    editorModel: EditorModel,
-    loader: EditableGridLoader,
+// TODO: we should refactor EditableGridLoader to make queryModel irrelevant. Some cases like AssayImportPanels create
+//  what is essentially a fake QueryModel just to satisfy this arg. EditableGridPanelForUpdateWithLineage passes the
+//  same QueryModel to every loader (it uses initEditorModels).
+export const initEditorModel = async (
     queryModel: QueryModel,
-    colFilter?: (col: QueryColumn) => boolean
-): Promise<{ dataModel: QueryModel; editorModel: EditorModel }> => {
-    const response = await loader.fetch(queryModel);
-    const gridData: Partial<QueryModel> = {
-        rows: response.data.toJS(),
-        orderedRows: response.dataIds.toArray(),
-        queryInfo: loader.queryInfo,
-    };
-
-    let columns: QueryColumn[];
+    loader: EditableGridLoader,
+    columnMetadata?: Map<string, EditableColumnMetadata>
+): Promise<EditorModel> => {
+    // TODO: look at EditorModel.getColumns and make sure we're matching the behavior from there as well e.g. file
+    //  columns
+    const { columns: loaderColumns, queryInfo } = loader;
+    // TODO: Most EditableGridLoaders do not actually asynchronously fetch data (see note in EditableGridLoader), we
+    //  can most likely just drop loader as an arg, and have consumers pass in their QueryModel data (if any), and
+    //  skip having QueryModel as an arg.
+    const { data, dataIds } = await loader.fetch(queryModel);
+    // TODO: store rows and orderedRows on EditorModel so we can use it with getUpdatedDataFromEditableGrid
```

Comment:
are any of these TODOs things you plan to look into for this PR?

Reply:
Yes, basically all of them.